src/server/services/BruincastServices.js

```diff
@@ -140,6 +140,114 @@ class BruincastServices {
     return formattedMedia;
   }
 
+  static async getAnalytics(
+    course,
+    members,
+    crosslistsCollectionName,
+    castCollectionName,
+    playbacksCollectionName
+  ) {
+    const labelList = await this.getCrosslistByCourse(
+      course.label,
+      crosslistsCollectionName
+    );
+    const courseList = [course.label, ...labelList];
+
+    // The following codes format analytics into the following format
+    // Top level (Course Level): { course: object, this course's analytics: array of second level }
+    // Second level (User Level):
+    //   { userid: number, name: string, this user's analytics: array of third level, finishedCount: number, totalCount: number }
+    // Third level (Media Level): { title of media: string, finishedTimes: number, time: number, remaining: number }
```

Comment:
@maxpluez It might also be helpful to add comments on certain lines in the code itself to briefly explain what's happening? (e.g. perhaps commenting where each level is being processed)